// Copyright (c) Huawei Technologies Co., Ltd. 2023. All rights reserved.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

message FfrtProfilerConfig {
    // Data collection when the program is running
    repeated int64 pid = 1;

    // Data collection in program startup stage
    repeated string startup_process_name = 2;

    // Data collection of program restart.
    repeated string restart_process_name = 3;

    enum ClockId {
        UNKNOW = 0;
        BOOTTIME = 1;
        REALTIME = 2;
        REALTIME_COARSE = 3;
        MONOTONIC = 4;
        MONOTONIC_COARSE = 5;
        MONOTONIC_RAW = 6;
    };
    ClockId clock_id = 4;
    uint32 smb_pages = 5;
    uint32 flush_interval = 6;

    // false: Discard data when there is no space in shared memory.
    // true: Block waiting when there is no space in the shared memory.
    bool block = 7;
}